wgpu-bindings/bindings.h

```diff
@@ -0,0 +1,112 @@
+#include <cstdint>
+#include <cstdlib>
+
+enum class PowerPreference {
+  Default = 0,
+  LowPower = 1,
+  HighPerformance = 2,
+};
+
+template<typename B>
+struct Buffer;
+
+template<typename B>
+struct CommandBuffer;
+
+template<typename B>
+struct ComputePass;
+
+template<typename B>
+struct Device;
+
+template<typename B>
+struct RenderPass;
+
+template<typename B>
+struct ShaderModule;
+
+struct ShaderModuleDescriptor;
+
+template<typename T>
+struct Handle {
+  T *_0;
+};
+
+using DeviceHandle = Handle<Device<B>>;
```

Comment:
Do we expect to receive pointers or handle structs for these? For example, if we want this to exactly match the API used for IPC, pointers might be better

Reply:
not sure I understand the question, please clarify

Reply:
Just wondering if the intent would be to reuse this header as the interface for IPC too (not sure how it's planned), or whether this header would only used directly (within the graphics process) with a separate interface for IPC.

Reply:
It's a great question! Here is the thing: we need wgpu-native C header so that native apps (written in C or anything) can target both native and WASM. And we need wgpu-remote C header so that Gecko can call into us. Technically, these are two completely different things, but I believe we'd have a much easier time if these were exactly the same. So I think we should use the header generated here for both.